code/src/utils/validators/productScheme.ts

```diff
@@ -0,0 +1,91 @@
+import { z } from "zod";
+
+import { ProductTranslation } from "@/containers/forms/new-product-form/NewProductForm.types";
+
+const validateProductLanguageFields = (data: ProductTranslation) => {
+  const isDescriptionEmpty = !data.description?.trim();
+  const isNameEmpty = !data.name?.trim();
+
+  const isError =
+    (!isNameEmpty && isDescriptionEmpty) ||
+    (!isDescriptionEmpty && isNameEmpty);
+
+  return !isError;
```

Comment:
Can you please explain how it works? Mutation tests will like this code, for sure :)

I also have a suggestion: 

```tsx
const isError = isNameEmpty != isDescriptionEmpty;
```

It is an example of `xor` operation in practice

Reply:
I needed some validation to ensure that each language's name and description were filled in. That is a function to validate such cases. It is used inside of `refine` method provided by `zod` validation schema.
As for mutation tests, I am sure some mutants unfortunately were able to survive, but I don't think that is a problem because validation of forms is tested in its test files and by me manually. I thought to exclude `validators` folder from stryker config. What do you think about it?
I also simplified that part according to your suggestion.

Reply:
I did not came up with testing `zod` validators, but I think we can exclude it from testing as we did with redux slices :)